### DISCUSSION TITLE
Limit based on response code

### DISCUSSION THREAD

**Discussion by vallyian** [nfriedly/express-rate-limit#317] [Ideas]
Could track 400 to prevent forms brute force
Could track 401/403 to prevent auth brute force
Could track 500+ to prevent DDNS attacks based on exception flow (most devs throw to interrupt regular flows and this usually requires a lot more resources to complete)

**Comment by nfriedly** [comment]
There's a couple of ways you could make this work with the current release of express-rate-limit:

- If you're OK with grouping all of the errors together, then just set the `skipSuccessfulRequests` flag, and it will only block users who hit too many errors.
- If you want to have each error counted separately, then create multiple instances of the rate-limiter with a custom `skip` function for each that skips everything except the one error it's trying to limit

And, in either case, if you want a different limit applied to successful requests, then set up one more instance for that.

If you're using a third-party store, then you'll likely need to configure it to use different keys for each instance of express-rate-limit.

---

In general, though, I wouldn't worry too much about this. Error codes, especially in the 4xx range, typically involve the server performing _less_ work than a successful request. So, a single all-purpose rate-limiter that applies to both successful and error requests is good enough to prevent abuse in most scenarios.
